src/functions.jl

```diff
@@ -409,3 +409,9 @@ LinearAlgebra.cholesky(A::Hermitian{T, <:KeyedArray{T}}; kwargs...) where {T} =
     cholesky(parent(A); kwargs...)
 LinearAlgebra.cholesky(A::KeyedMatrix; kwargs...) =
     cholesky(keyless_unname(A); kwargs...)
+
+function Base.deleteat!(v::KeyedVector, inds)
+    deleteat!(v.data, inds)
+    deleteat!(v.keys, inds)
```

Comment:
I don't think this is the right way to do it
```julia
julia> kv = KeyedArray([1, 2, 3, 4, 5, 6.0], a=[:a, :b, :c, :d, :e, :f]);

julia> kv.keys
Base.RefValue{Vector{Symbol}}([:a, :b, :c, :d, :e, :f])

julia> deleteat!(kv.keys, 1)
ERROR: MethodError: no method matching deleteat!(::Base.RefValue{Vector{Symbol}}, ::Int64)
Closest candidates are:
  deleteat!(::Vector{T} where T, ::Integer) at array.jl:1343
  deleteat!(::Vector{T} where T, ::Any) at array.jl:1380
  deleteat!(::BitVector, ::Integer) at bitarray.jl:949
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[29]:1
```

I get the same error when I call `deleteat!(kv, 1)`.
I think instead  you need
```suggestion
    deleteat!(axiskeys(v, 1), inds)
```

This fixes it on my end.

Reply:
Oh right, thanks! I implemented this before the NamedDims PR, and then assumed the CI is broken because NamedDims is not registered yet 🤦

Reply:
Since the key vector seems more likely to be immutable, e.g. a range, maybe this should try that first, so that it is more likely not to leave an invalid state if it fails?

`v.keys` is in fact a `Ref(vec)`, so some immutables could in fact be replaced. But not ranges, except at the end. Maybe not worth trying.

Reply:
good point - I've changed the order and added a test for that